mill-scalafix/src/com/goyeau/mill/scalafix/ScalafixModule.scala

```diff
@@ -71,7 +73,7 @@ object ScalafixModule {
         new ScalafixInterfacesClassloader(getClass.getClassLoader)
       )
       val scalafix = Scalafix
-        .classloadInstance(toolClassloader)
+        .fetchAndClassloadInstance("2.12", repositories.map(CoursierUtils.repository).asJava)
```

Comment:
By using this helper instead of your local own classloader, https://github.com/scalacenter/scalafix/blob/master/scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixArguments.java#L70-L72 is violated later.

The new APIs are meant to abstract completely the artifact fetching & classloading hierarchy, so that you can't end up with incompatible Scala binary versions for example (the version provided here should control everything else). However, you have to use the new higher-level `withToolClasspath` overloads. I left the low-level classloader overloads for backward compatibility, and didn't deprecate them as they can serve some very advanced scenarios, but their usage should really be exceptional. I will add some comments in the Javadoc to clarify.

In this client, that means `ScalafixInterfacesClassloader` and https://github.com/joan38/mill-scalafix/blob/48810b5e323ce16177ad00b85222f4e5be8e4222/mill-scalafix/src/com/goyeau/mill/scalafix/ScalafixModule.scala#L37-L42 should disappear.

Have look at https://github.com/scalacenter/sbt-scalafix/pull/121/commits/d81dd7266231e70af475551b563b21ec75c89482 if that's not clear.